src/detypePlugin.ts

```diff
@@ -0,0 +1,174 @@
+    const codeBlockClose = '```'
+
+    // Remove TypeScript
+    if (!isYaml) {
+      codeBlockContentJs = await detype(codeBlockContentJs, `some-dummy-filename.${codeBlockLang}`, {
+        customizeBabelConfig(config) {
+          // Add `onlyRemoveTypeImports: true` to the internal `@babel/preset-typescript` config
+          // See https://github.com/cyco130/detype/blob/main/src/transform.ts#L206
+          assertUsage(config.presets && config.presets.length === 1, 'Unexpected Babel config presets')
+          config.presets = [[config.presets[0], { onlyRemoveTypeImports: true }]]
+        },
+        removeTsComments: true,
+        prettierOptions,
+      })
+      // Correct code diff comments
+      codeBlockContentJs = correctCodeDiffComments(codeBlockContentJs)
+    }
+
+    if (codeBlockContentJs === codeBlockContent) {
+      continue
+    } else {
+      // Update code block open delimiter
+      const codeBlockLangJs =
+        codeBlockLang === 'vue'
+          ? 'vue'
+          : // ts => js | tsx => jsx
+            codeBlockLang.replace('t', 'j')
+      const codeBlockOpenJs = codeBlockOpen.replace(codeBlockLang, codeBlockLangJs)
+
+      const codeBlockJs = `${codeBlockOpenJs}\n${codeBlockContentJs}${codeBlockClose}`
+      const codeBlockTs = `${codeBlockOpen}\n${codeBlockContent}${codeBlockClose}`
+
+      // Combine JS & TS code blocks
+      let codeSnippets = `${codeBlockJs}\n${codeBlockTs}`
+
+      if (!isYaml) {
+        // Rename/Replace Words via Custom Magic Comments
+        codeSnippets = processMagicComments(codeSnippets)
+      }
+
+      // Wrap with <CodeSnippets>
+      codeSnippets = `<CodeSnippets>\n${codeSnippets}\n</CodeSnippets>`
+
+      // Restore indentation
+      codeSnippets = restoreCodeBlockIndent(codeSnippets, codeBlockIndent)
+
+      // Done
+      replacement = codeSnippets
+    }
+
+    const blockStartIndex = match.index!
+    const blockEndIndex = blockStartIndex + codeBlockOuterStr.length
+    magicString.overwrite(blockStartIndex, blockEndIndex, replacement)
+  }
+
+  return getMagicStringResult()
+}
+
+function removeCodeBlockIndent(code: string, codeBlockIndent: string, moduleId: string) {
+  if (!codeBlockIndent.length) return code
+  return code
+    .split('\n')
+    .map((line) => {
+      const lineStart = codeBlockIndent.trimEnd()
+      assertUsage(
+        line.startsWith(lineStart),
+        `In ${pc.bold(pc.blue(moduleId))} the line '${pc.bold(line)}' must start with '${pc.bold(lineStart)}'`,
+      )
+      return line.slice(codeBlockIndent.length)
+    })
+    .join('\n')
+}
+function restoreCodeBlockIndent(code: string, codeBlockIndent: string) {
+  if (!codeBlockIndent.length) return code
+  return code
+    .split('\n')
+    .map((line) => `${codeBlockIndent}${line}`)
+    .join('\n')
+}
+function processMagicComments(code: string) {
+  // @detype-rename DummyLayout>Layout
+  const renameCommentRE = /^\/\/\s@detype-rename\s(\w+)>(\w+)\n/gm
```

Comment:
How about this? It's easier to read and more conventional. I don't expect any use case for replacing a name containing a white space.
```suggestion
  const renameCommentRE = /^\/\/\s@detype-rename\s([^ ]+) ([^ ]+)\n/gm
```